lib/chef-cli/command/gem.rb

```diff
@@ -29,17 +30,51 @@ module Command
     class GemForwarder < ChefCLI::Command::Base
       banner "Usage: #{ChefCLI::Dist::EXEC} gem GEM_COMMANDS_AND_OPTIONS"
 
+      option :skip_rehash,
+        long:        "--skip-rehash",
+        description: "Don't run rehash the cache of knife plugins after running the gem command.",
+        default:     false
+
       def run(params)
-        retval = Gem::GemRunner.new.run( params.clone )
+        retval = Gem::GemRunner.new.run( sanitized_params(params.clone) )
         retval.nil? ? true : retval
-      rescue Gem::SystemExitException => e
+      rescue Gem::SystemExitException => e # yes we're using a rescue for flow control. :(
+        unless skip_rehash?
+          Chef::Knife::SubcommandLoader.write_hash(Chef::Knife::SubcommandLoader.generate_hash)
```

Comment:
Since these are often called together, making a simpler interface could insulate against changes in the future:

```suggestion
          Chef::Knife::SubcommandLoader.rehash
```